granadilla/forms.py

```diff
@@ -48,6 +49,116 @@ class Meta:
         model = LdapContact
         exclude = ("dn", "photo", "full_name")
 
+class LdapDeviceForm(forms.ModelForm):
+
+    password2 = forms.CharField(widget=forms.PasswordInput, label = "Re-Password")
+
+    def __init__(self, *args, **kwargs):
+        if 'base_dn' in kwargs:
+            self.__base_dn = kwargs.pop('base_dn')
+        else:
+            self.__base_dn = None
+        self.__username = kwargs.pop('username')
+        super(LdapDeviceForm, self).__init__(*args, **kwargs)
+
+
+    def is_valid(self):
+        if super(LdapDeviceForm, self).is_valid():    
+            # Check if the device already exists
+            device_fullname = self.__username + '_' + \
+                        self.data['device_name']
+            try:
+                if models.LdapDevice.objects.get(device_fullname=device_fullname):
+                    self.errors['devicenameerror'] = "A device with \
+                            this name already exists"
+                    return False
+            except models.LdapDevice.DoesNotExist:
```

Comment:
Nitpicking: you could just add a ``pass`` after this block, and unindent the code below; this would be easier to read :)